### PR TITLE
Add ecr-credential-helper to docker image

### PR DESCRIPTION
**Pull request by kon-angelo**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds ecr-credential-provider to the aws-cloud-controller-manager image.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/cloud-provider-aws/issues/823

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/ecr-credential-provider/plugin/plugin.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package plugin
```
Comment (cartermckinnon) [inline cmd/ecr-credential-provider/plugin/plugin.go:17]: This change seems unrelated, can you revert it if it's not necessary for this PR?

**Comment by cartermckinnon** [comment]
/hold

While we explore other options. I don't think this is totally unreasonable, but I don't think it'll be used very often; and it has the downsides of increased image size and surface area for security scanners.

File: cmd/ecr-credential-provider/plugin/plugin.go (hunk 1 above)
Comment (kon-angelo) [reply cmd/ecr-credential-provider/plugin/plugin.go:17]: This solves a very peculiar thing I saw. Without it, compilation (aka `go build`) would fail if not performed inside `cmd/ecr-credential-provider` directory (requiring you to `cd`) and the error was that the `NewCredentialProvider` could not be found.

It all goes away when you refactor the `plugin.go` file out of the `main` package though. You can "script" around the issue but refactoring to a proper package seemed the better way to go. We can argue about the package location of course.

**Comment by hakman** [comment]
> While we explore other options. I don't think this is totally unreasonable, but I don't think it'll be used very often; and it has the downsides of increased image size and surface area for security scanners.

@cartermckinnon I suggest creating a separate image for it, in case this is something needed in the future. I don't see any value in bundling it in a single image.

**Comment by cartermckinnon** [comment]
> I suggest creating a separate image for it, in case this is something needed in the future. I don't see any value in bundling it in a single image.

Agreed; I don't think we're going to go this route. A container image just isn't the right way to distribute this.

/close
